### ISSUE TITLE
Unable to load class dictionary for `InDetTrackTruthFilterTool`

### ISSUE THREAD

**Issue by kgreif24** [UCATLAS/xAODAnaHelpers#1635]
Hello,

I have been using xAODAnaHelpers to produce nTuples of jet information. I am compiling against AnalysisBase 21.2.180. Recently I've been trying to apply a track systematic tool, for which I have been following this helpful documentation: https://xaodanahelpers.readthedocs.io/en/latest/Development.html#adding-and-initializing-tools

When I follow this prescription for using the `AnaToolHandle` class, the code compiles but at runtime I get an error in the call to the `retrieve` method of the handle class:

```
Package.AsgTools.AnaTo...ERROR   /build/atnight/localbuilds/nightlies/AnalysisBase/21.2/athena/Control/AthToolSupport/AsgTools/Root/AnaToolHandle.cxx:157 (StatusCode asg::detail::makeToolRootCore(const string&, const string&, asg::AsgTool*&)): Unable to load class dictionary for type InDetTrackTruthFilterTool
```

This error appears to be coming from [this location](https://gitlab.cern.ch/atlas/athena/-/blob/21.2/Control/AthToolSupport/AsgTools/Root/AnaToolHandle.cxx#L150-160) in the handle class code, but I am unsure exactly what this code is doing. Based on the documentation, this could be an issue with the tool header file versus an issue with xAODAnaHelpers. If this the case please let me know. If not I wonder if I am either doing something wrong in following the documentation, or if there is some part of the `AnaToolHandle` class that is currently not supported.

For reference, here are the links to the relevant code in my algorithm:
- Header file: https://gitlab.cern.ch/kgreif/daod-dumper-with-systematics/-/blob/master/TopWBosonSubstructure/TopWBosonSubstructure/TopWBosonSubstructureTreeAlgo.h#L59-62
- Implementation: https://gitlab.cern.ch/kgreif/daod-dumper-with-systematics/-/blob/master/TopWBosonSubstructure/Root/TopWBosonSubstructureTreeAlgo.cxx#L64-68

Thank you in advance for the help!

**Comment by kratsg** [comment]
Make sure you update your `LinkDef.h` file as well.

**Comment by kgreif24** [comment]
After some work, I found that the issue was with another tool called by the tool I want to run. Namely the `InDetTrackTruthFilterTool` requires the `InDetTrackTruthOriginTool` to also be initialized properly. I made this second tool a public member of my algorithm class, following the prescription described in the documentation. After doing this, everything works as expected. Therefore marking the issue closed.
